src/geff/networkx/io.py

```diff
@@ -150,7 +157,30 @@ def _set_property_values(
                 graph.edges[source, target][name] = val
 
 
-def read_nx(path: Path | str, validate: bool = True) -> nx.Graph:
+def _ingest_dict_nx(graph_dict: GraphDict):
+    metadata = graph_dict["metadata"]
+
+    graph = nx.DiGraph() if metadata.directed else nx.Graph()
+    for key, val in metadata:
+        graph.graph[key] = val
```

Comment:
This will conflict with the latest main, where we have removed storing attributes on the graph and instead just return the metadata with the graph